src/client/Microsoft.Identity.Client/Extensibility/CdtAuthenticationScheme.cs

```diff
@@ -0,0 +1,296 @@
+using JObject = System.Text.Json.Nodes.JsonObject;
+using JToken = System.Text.Json.Nodes.JsonNode;
+#else
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Json.Linq;
+#endif
+
+namespace Microsoft.Identity.Client
+{
+    //Temporary location
+    public class CdtAuthenticationScheme : IAuthenticationScheme
+    {
+        //CDT
+        public const string CdtKey = "xms_ds_cnf";
+        public const string CdtNonce = "xms_ds_nonce";
+        public const string CdtEncKey = "xms_ds_enc";
+        public const string NoAlgorythmPrefix = "none";
+        public const string JasonWebTokenType = "JWT";
+        public const string CdtTokenType = "CDT";
+        public const string CdtEncryptedAlgoryth = "dir";
+        public const string CdtEncryptedValue = "A256CBC-HS256";
+        public const string CdtRequestConfirmation = "req_ds_cnf";
+
+        private readonly CdtCryptoProvider _cdtCryptoProvider;
+        private readonly string _constraints;
+        private readonly string _dsReqCnf;
+
+        /// <summary>
+        /// Creates Cdt tokens, i.e. tokens that are bound to an HTTP request and are digitally signed.
+        /// </summary>
+        /// <remarks>
+        /// Currently the signing credential algorithm is hard-coded to RSA with SHA256. Extensibility should be done
+        /// by integrating Wilson's SigningCredentials
+        /// </remarks>
+        public CdtAuthenticationScheme(string constraints, X509Certificate2 certificate)
+        {
+            _constraints = constraints ?? throw new ArgumentNullException(nameof(constraints));
+
+            _cdtCryptoProvider = new CdtCryptoProvider(certificate);
+
+            _dsReqCnf = _cdtCryptoProvider.CannonicalPublicKeyJwk;
+        }
+
+        public TokenType TelemetryTokenType => TokenType.Bearer;
+
+        public string AuthorizationHeaderPrefix => Constants.BearerAuthHeaderPrefix;
+
+        public string AccessTokenType => Constants.BearerAuthHeaderPrefix;
+
+        /// <summary>
+        /// For Cdt, we chose to use the base64(jwk_thumbprint)
+        /// </summary>
+        public string KeyId { get; }
+
+        public IReadOnlyDictionary<string, string> GetTokenRequestParams()
+        {
+            var temp = "eyJrdHkiOiAiUlNBIiwgIm4iOiAiMUNMeDNXRW1NWlQ3el92Szc2ZHBaVVNwX2kyMEEza0Y0OWVpemtCQTBFSTJ4el9pcldTcm9BamJrRTk4dlp3SFM0QVlQV2I5WEd2YTFPYVNMX0RqQTFPTG1nSll4Uk45cU5jd1lKeGhsN3hqaGJlU25RMUMtR1NNS3ZWRzJnaDdQUlhaaU1xVXFuOWt3UzBXa1RoNDhSREMxR0xhTFFfNzZmb0dZMmo0MlNvel9XYnNRemtnVGo0TDVaVTZTWjJ3QTFwMlZ6WFliOVd1M3A4U2VuV3JCTDUzOWhUZjVGelp0b1E0R2IxNzMzVzFmWVFsUkotYUZVMTFfdEc1Umx2Ui1nSWFweHJMWkFKM1NHM28wQ2ZPa2FaejdKT2RETnJHNnE4akF3ZmJOdFJ1eDYzbnJZZ0FHc3VhemlXalZxRnZiclNMX2Mya3dZaDlZUl9uYVJFOG1RPT0iLCAiZSI6ICJBUUFCIn0%3D";
+            return new Dictionary<string, string>() {
+                { OAuth2Parameter.TokenType, Constants.BearerAuthHeaderPrefix},
+                { CdtRequestConfirmation, Base64UrlHelpers.Encode(_dsReqCnf)}
```

Comment:
`Base64UrlHelpers` not public and cannot be used in an extension either.

Reply:
What do you suggest? Should we add this helper to Abstractions? Base64UrlHelper is in all our libs.

Reply:
Wilson has the necessary helper

```csharp
string base64UrlEncoded= Base64UrlEncoder.Encore(decoded );
```

It's in Microsoft.IdentityModel.Tokens